### PR TITLE
[Scheduled-Query] update api version

### PR DESCRIPTION
**Pull request by kairu-ms**
closed https://github.com/Azure/azure-cli/issues/17891

---

* Add `--skip-query-validation` parameter
* Add `--check-ws-alerts-storage` parameter
* [Breaking Change] the default value of `--mute-actions-duration` is removed because alert auto mitigate is supported

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Scheduled-Query

**Comment by yanivlavi** [comment]
@kairu-ms 
Flag '--mute-actions-duration' is not removed.
It is only supported when stateful is disabled. Stateful should be a flag control setting with default being enabled.

Please pass this PR with us for sign-off before you merge.

**Comment by yanivlavi** [comment]
Please make sure you do not diverge from metric alerts as they also have 'auto-mitigate' option.
Keep in mind we want to be aligned with them.

**Comment by kairu-ms** [comment]
> @kairu-ms
> Flag '--mute-actions-duration' is not removed.
> It is only supported when stateful is disabled. Stateful should be a flag control setting with default being enabled.
> 
> Please pass this PR with us for sign-off before you merge.

Thanks for your feedback, I have two question:
  1. What do you mean by stateful? Can user enable or disable stateful by cli?
  2. Do you want to remove `--mute-actions-duration` flag?

**Comment by yanivlavi** [comment]
> Thanks for your feedback, I have two question:
> 
> 1. What do you mean by stateful? Can user enable or disable stateful by cli?

Yes, it is controlled by the 'autoMitigate' flag in the API. Would like to have that controllable in the CLI.

> 2. I don't remove the `--mute-actions-duration` flag. I've changed the default value of it to None.  Do you want to remove it?

No, when 'autoMitigate' is false you can still set mute-actions-duration. 
The defaults should be 'autoMitigate' true, mute-actions-duration is fine to be None as default.

**Comment by yanivlavi** [comment]
What was the default for muting before?

**Comment by kairu-ms** [comment]
> What was the default for muting before?

The default value is 30 minutes

**Comment by kairu-ms** [comment]
> > Thanks for your feedback, I have two question:
> > 
> > 1. What do you mean by stateful? Can user enable or disable stateful by cli?
> 
> Yes, it is controlled by the 'autoMitigate' flag in the API. Would like to have that controllable in the CLI.
> 
> > 1. I don't remove the `--mute-actions-duration` flag. I've changed the default value of it to None.  Do you want to remove it?
> 
> No, when 'autoMitigate' is false you can still set mute-actions-duration.
> The defaults should be 'autoMitigate' true, mute-actions-duration is fine to be None as default.

I see. I will add that control in CLI.

File: src/scheduled-query/azext_scheduled_query/_params.py [hunk 1]
```diff
@@ -36,3 +38,6 @@ def load_arguments(self, _):
                    options_list=['--mute-actions-duration', '--mad'],
                    help='Mute actions for the chosen period of time (in ISO 8601 duration format) after the alert is fired.')
         c.argument('actions', options_list=['--action', '-a'], action=ScheduleQueryAddAction, nargs='+', validator=get_action_group_validator('actions'))
+        c.argument('disable_auto_mitigate', arg_type=get_three_state_flag(), help='The flag that indicates whether to disable the alert automatically resolved or not.')
```
Comment (jsntcy) [inline src/scheduled-query/azext_scheduled_query/_params.py:41]: From swagger, it's defined as `auto-mitigate` with default value `true`, could you please explain the purpose that define it as `disable-auto-mitigate`

File: src/scheduled-query/azext_scheduled_query/_params.py [hunk 2]
```diff
@@ -36,3 +38,6 @@ def load_arguments(self, _):
...
+        c.argument('skip_query_validation', arg_type=get_three_state_flag(), help='The flag which indicates whether the provided query should be validated or not.')
+        c.argument('check_workspace_alerts_storage', options_list=['--check-ws-alerts-storage', '--cwas'], arg_type=get_three_state_flag(), help="The flag which indicates whether this scheduled query rule should be stored in the customer's storage.")
```
Comment (jsntcy) [inline src/scheduled-query/azext_scheduled_query/_params.py:43]: Please make the help more detailed as swagger does including what kind of alert it applies, default valule.

File: src/scheduled-query/azext_scheduled_query/_params.py [hunk 3]
```diff
@@ -36,3 +37,6 @@ def load_arguments(self, _):
                    options_list=['--mute-actions-duration', '--mad'],
                    help='Mute actions for the chosen period of time (in ISO 8601 duration format) after the alert is fired.')
         c.argument('actions', options_list=['--action', '-a'], action=ScheduleQueryAddAction, nargs='+', validator=get_action_group_validator('actions'))
+        c.argument('auto_mitigate', arg_type=get_three_state_flag(), help='The flag that indicates whether the alert should be automatically resolved or not. The default is true.')
+        c.argument('skip_query_validation', arg_type=get_three_state_flag(), help='The flag which indicates whether the provided query should be validated or not.')
+        c.argument('check_workspace_alerts_storage', options_list=['--check-ws-alerts-storage', '--cwas'], arg_type=get_three_state_flag(), help="The flag which indicates whether this scheduled query rule should be stored in the customer's storage.")
```
Comment (jsntcy) [inline src/scheduled-query/azext_scheduled_query/_params.py:42]: Add default value in help message

**Review by jsntcy — APPROVED** [review]
:shipit:

**Comment by kairu-ms** [comment]
Hi, @yanivlavi does this PR ready to release?

**Pull request merged**